src/Dapr.Actors/Runtime/ActorSettings.cs

```diff
@@ -0,0 +1,109 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Actors.Runtime
+{
+    using System;
+    using System.Text.Json;
+
+    /// <summary>
+    /// Represents Dapr actor configuration for this app.  See
+    /// https://github.com/dapr/docs/blob/master/reference/api/actors_api.md
+    /// </summary>
+    public sealed class ActorSettings
+    {
+        /// <summary>
+        /// Default constructor.  This is only here for the unit test.
+        /// </summary>
+        private ActorSettings()
+            : this(null, null, null, false)
+        { }
+
+        /// <summary>
+        /// Users should use this constructor to override default values.
+        /// </summary>
+        /// <param name="actorIdleTimeout">Specifies how long to wait before deactivating an idle actor. An actor is idle if no actor method calls and no reminders have fired on it.</param>
+        /// <param name="actorScanInterval">A duration which specifies how often to scan for actors to deactivate idle actors. 
+        /// Actors that have been idle longer than the actorIdleTimeout will be deactivated.</param>
+        /// <param name="drainOngoingCallTimeout">A duration used when in the process of draining rebalanced actors. This specifies 
+        /// how long to wait for the current active actor method to finish. If there is no current actor method call, this is ignored.</param>
+        /// <param name="drainRebalancedActors">A bool. If true, Dapr will wait for drainOngoingCallTimeout to allow a current 
+        /// actor call to complete before trying to deactivate an actor. If false, do not wait.</param>
+        public ActorSettings(
+            TimeSpan? actorIdleTimeout = null,
```

Comment:
Don't add this constructor, let users set the values through Properties, move the validations to setter of properties.
If we change the constructor to accept more values or change the order in such a way that its not a compile time error for users, it will cause issues.